app/services/position_adjuster_inside_list.rb

```diff
@@ -0,0 +1,49 @@
+class PositionAdjusterInsideList
+
+  def initialize(list)
+    @list = list
+  end
+
+  def collection
+    @list.todopoints
+  end
+
+  def move(direction, todo_id)
+    if direction == :up
+      move_up(todo_id)
+      true
+    elsif direction == :down
+      move_down(todo_id)
+      true
+    end
+  end
+
+  def move_up(todo_id)
+    todo_which_i_want_to_move_up = collection.find_by(id: todo_id)
```

Comment:
nie powinieneś stosować find_by(id: ), bo to może zwrócić nil'a, którego się nie spodziewasz (todo_which_i_want_to_move.position zwróci Ci wyjątek). zamień na find_by!(id: ) a właściwie na jego równoważnik czyli find(id)